typed-racket-lib/typed-racket/rep/type-mask.rkt

```diff
@@ -149,6 +149,8 @@
  mask:struct
  mask:prefab
  mask:struct-type
+ mask:self
+ mask:struct-property
  mask:syntax
```

Comment:
I haven't put much thought into the validity of `mask:struct-property`.... I could believe it's disjoint from all the other descriptors. Might want to double check that.

Also, is it worth having it's own type mask? I do not know, but I just feel like they'll probably not be frequently encountered in type computations (but I could be wrong).